gcm/gcm.py

```diff
@@ -278,30 +352,41 @@ def plaintext_request(self, **kwargs):
         """
         Makes a plaintext request to GCM servers
 
-        :param registration_id: string of the registration id
-        :param data: dict mapping of key-value pairs of messages
         :return dict of response body from Google including multicast_id, success, failure, canonical_ids, etc
         """
 
         kwargs['is_json'] = False
         retries = kwargs.pop('retries', 5)
         payload = self.construct_payload(**kwargs)
 
-        attempt = 0
         backoff = self.BACKOFF_INITIAL_DELAY
+        info = None
+        has_error = False
 
-        if retries:
-            for attempt in range(retries):
-                try:
-                    response = self.make_request(payload, is_json=False)
-                    return self.handle_plaintext_response(response)
-                except GCMUnavailableException:
-                    sleep_time = backoff / 2 + random.randrange(backoff)
-                    time.sleep(float(sleep_time) / 1000)
-                    if 2 * backoff < self.MAX_BACKOFF_DELAY:
-                        backoff *= 2
+        for attempt in range(retries):
+            try:
+                response = self.make_request(payload, is_json=False)
+                info = self.handle_plaintext_response(response)
+                has_error = False
+            except GCMUnavailableException:
+                has_error = True
+
+            if self.retry_after:
+                GCM.log("[Attempt #{0}] Retry-After ~> Sleeping for {1} seconds".format(attempt, self.retry_after))
+                time.sleep(self.retry_after)
```

Comment:
`time.sleep` can't take a `None` value, shouldn't you have some error handling around this?